pai-management/paictl.py

```diff
@@ -163,26 +165,38 @@ def cluster_object_model_generate_service(config_path):
 
 
 
-def hadoop_ai_build(os_type = "ubuntu16.04"):
+def pai_build():
 
     parser = argparse.ArgumentParser()
     parser.add_argument('-p', '--path', required=True, help="The path of your configuration directory.")
+    parser.add_argument('-n', '--imagename', default='all', help="Build and push the target image to the registry")
+    parser.add_argument('-a', '--action', default=None, help="Only build or push" )
     args = parser.parse_args(sys.argv[1:])
 
     config_path = args.path
+    image_name = args.imagename
     cluster_object_model = load_cluster_objectModel_service(config_path)
 
-    hadoop_path = cluster_object_model['clusterinfo']['hadoopinfo']['custom_hadoop_binary_path']
+    image_list = None
+    if image_name != "all":
+        image_list = [ image_name ]
 
-    commandline = "./paiLibrary/managementTool/{0}/hadoop-ai-build.sh {1}".format(os_type, hadoop_path)
-    error_msg = "Failed to build hadoop-ai."
-    linux_shell.execute_shell(commandline, error_msg)
+    if args.action != None and args.action not in ["build", "push"]:
+        logger.error("The option -a {0} is invalid.".format(args.action))
+        logger.error("Only build image please type   : paictl.py image -p /path/to/configuration/ -a build [ -n image-x ]")
+        logger.error("Only push image please type    : paictl.py image -p /path/to/configuration/ -a push [ -n image-x ]")
+        logger.error("Build & push image please type : paictl.py image -p /path/to/configuration/ [ -n image-x ]")
+        return
 
+    if args.action == None or args.action == "build":
+        center = build_center.build_center(cluster_object_model, image_list)
```

Comment:
>cluster_object_model [](start = 43, length = 20)

Will we seperate cluster config from image the next version?

Reply:
Yes.